Public/Src/Sandbox/MacOs/Interop/Posix/memory.c

```diff
@@ -42,15 +42,58 @@ int GetRamUsageInfo(RamUsageInfo *buffer, long bufferSize)
     return KERN_SUCCESS;
 }
 
+static uint64_t ProcessTreeWorkingSetSize(pid_t pid, const rlim_t max_proc_count, bool *success)
+{
+    uint64_t children_usage = 0;
+
+    pid_t pids[max_proc_count];
+    int child_count = proc_listchildpids(pid, pids, (int) max_proc_count);
+    *success &= child_count >= 0;
+
+    for (int i = 0; (i < child_count) && *success; i++)
+    {
+        int pid = pids[i];
+        rusage_info_current rusage;
+        *success &= proc_pid_rusage(pid, RUSAGE_INFO_CURRENT, (void **)&rusage) == 0;
```

Comment:
>RUSAGE_INFO_CURRENT, [](start = 41, length = 20)

This data is exposed by GetPeakWorkingSetSize. Is it really peak or is it current?

Reply:
It's current really - looking at the code in general it seemed to me like  "peak" is used by the job object as it is the only readable information. In the context this is called BuildXL is actually doing calculations to free RAM and get below a threshold. Do you think peak is what we should use?

Reply:
The windows code is using peak for historical runtime tracking. I'm not sure for cancellation. I guess for cancellation of the currently running pips it would be best to have max of historical peak and current. I'd mirror the windows implementation for our own sanity. But doing the peak is probably best if we had to be pick one.

Reply:
We do this already, although the actual number is the current process tree memory size, the `AccountingInformation` for a sandboxed process returns `PeakMemoryUsage = (ulong)m_perfAggregator.PeakMemoryBytes.Maximum` so this works as expected because that value is used for the historic data.